src/scripts/main.js

```diff
@@ -1,3 +1,179 @@
 'use strict';
 
-// write your code here
+const score = document.querySelector('.game-score');
+const initializeGameBtn = document.querySelector('.button.start');
+const gameOver = document.querySelector('.message-lose');
+const youWin = document.querySelector('.message-win');
+const startMessage = document.querySelector('.message-start');
+const rows = Array.from(document.querySelectorAll('.field-row'));
+const board = Array(4).fill(0).map(x => Array(4).fill(0));
+
+function findEmptyCell() {
+  const boardSize = {
+    row: 0,
+    column: 0,
+  };
+
+  do {
+    boardSize.row = Math.floor(Math.random() * 4);
+    boardSize.column = Math.floor(Math.random() * 4);
+  } while (board[boardSize.row][boardSize.column] !== 0);
+
+  return boardSize;
+}
+
+function addNumber() {
+  const { row, column } = findEmptyCell();
+
+  board[row][column] = Math.random() < 0.9 ? 2 : 4;
+}
+
+function render() {
+  board.map((row, rowIdx) => {
+    row.map((cell, columnIdx) => {
+      const elem = rows[rowIdx].children[columnIdx];
+
+      if (cell === 0) {
+        elem.textContent = '';
+        elem.classList = 'field-cell';
+      } else {
+        elem.textContent = cell;
+        elem.classList = `field-cell field-cell--${cell}`;
+      }
+    });
+  });
+
+  score.textContent = setScore();
+}
+
+function startGame() {
+  board.map(row =>
+    row.map((cell, cellIdx, cellArr) => (
+      cellArr[cellIdx] = 0)));
+
+  render();
+  startMessage.classList.add('hidden');
+  initializeGameBtn.classList.remove('start');
+  initializeGameBtn.classList.add('restart');
+  initializeGameBtn.textContent = 'Restart';
+  score.textContent = '0';
+
+  if (!youWin.classList.contains('hidden')) {
+    youWin.classList.add('hidden');
+  }
+
+  if (!gameOver.classList.contains('hidden')) {
+    gameOver.classList.add('hidden');
+  }
+
+  addNumber();
+  addNumber();
+  render();
+}
+
+function setScore() {
+  return board.reduce((prev, row) => (
+    prev + row.reduce((sum, cell) => sum + cell)
+  ), 0);
+}
+
+function normalizeArr(move, initialArray, useReverse = false) {
+  const normalizedArray = [[], [], [], []];
+
+  if (move === 'ArrowUp' || move === 'ArrowDown') {
```

Comment:
move 'directions' to object and use keys to get values


const KEY_DIRECTIONS = {
  UP: 'ArrowUp',
  etc...
}